src/axom/quest/readers/C2CReader.hpp

```diff
@@ -67,9 +67,25 @@ class C2CReader
    * 
    * Knot spans are the sub-intervals within a spline
    */
-  void getLinearMesh(mint::UnstructuredMesh<mint::SINGLE_SHAPE>* mesh,
+  void getLinearMesh(mint::UnstructuredMesh<mint::SINGLE_SHAPE> *mesh,
                      int segmentsPerKnotSpan);
 
+  /*!
+   * \brief Projects high-order NURBS contours onto a linear mesh using \a percentError 
+   *        to decide when to stop refinement. The method also computes an
+   *        approximation of the revolved volume that would be created by the
+   *        contours, using quadrature.
+   * 
+   * \param[in] mesh The mesh object that will contain the linearized line segments.
+   * \param[in] transform A transformation matrix that will be applied to points.
+   * \param[in] percentError A percent of error that is acceptable to stop refinement.
+   * \param[out] revolvedVolume The revolved volume of the curve (not linearized).
+   */
+  void getLinearMesh(mint::UnstructuredMesh<mint::SINGLE_SHAPE> *mesh,
+                     const numerics::Matrix<double> &transform,
+                     double percentError,
+                     double &revolvedVolume);
```

Comment:
On second thought, perhaps we should have a separate explicit function to compute the revolved volume, rather than folding it into this `getLinearMesh` function

Reply:
We can do that.

Reply:
I separated out the revolved volume calculation into a getRevolvedVolume() method.